src/git-extract

```diff
@@ -54,17 +54,18 @@ function preconditions {
 }
 
 
-function steps {
-  echo_if_true "stash_open_changes" "$INITIAL_OPEN_CHANGES"
+function should_stash_open_changes {
```

Comment:
I personally prefer opt in. Also makes it easier to tell from each commands perspective if its stashing or not with less knowledge of the framework. 

I think its makes more sense as a function. `git ship` needs a variable set by preconditions.